src/VCR/MatchObserver.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace VCR;
+
+class MatchObserver
+{
+    // This is an array keyed by a hash of the (first aka stored) request.
+    private $requestToMatchersMap;
+    private $requestCount0;
+    private $requestToCountMap;
+    private $requestToRequestsMap;
+    private $shouldObserve = false;
+
+    public function clear() {
+        $this->requestToMatchersMap = array();
+        $this->requestCount = 0;
+        $this->requestToCountMap = array();
+        $this->requestToRequestsMap = array();
+    }
+
+    public function markMismatch(Request $first, Request $second, $matcher) {
+        $hash = $first->getIdentityHash();
```

Comment:
I didn't want the order of matching to matter, so an identity hash was a way to keep matches with their request.